http/php/guzzle/src/GuzzleRequestAdapter.php

```diff
@@ -0,0 +1,244 @@
+        $this->guzzleClient = ($guzzleClient) ?: KiotaClientFactory::create();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function sendAsync(RequestInformation $requestInfo, string $targetClass, ?ResponseHandler $responseHandler = null): Promise
+    {
+        $response = $this->getHttpResponseMessage($requestInfo);
+        return $response->then(
+            function (ResponseInterface $result) use ($targetClass, $responseHandler) {
+                if (!$responseHandler) {
+                    $rootNode = $this->getRootParseNode($result);
+                    if ($targetClass === StreamInterface::class || in_array(StreamInterface::class, class_implements($targetClass))) {
+                        return $result->getBody();
+                    }
+                    return $rootNode->getObjectValue($targetClass);
+                }
+                return $responseHandler->handleResponseAsync($result);
+            }
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getSerializationWriterFactory(): SerializationWriterFactory
+    {
+        return $this->serializationWriterFactory;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function sendCollectionAsync(RequestInformation $requestInfo, string $targetClass, ?ResponseHandler $responseHandler = null): Promise
+    {
+        $response = $this->getHttpResponseMessage($requestInfo);
+        return $response->then(
+            function (ResponseInterface $result) use ($targetClass, $responseHandler) {
+                if (!$responseHandler) {
+                    $rootNode = $this->getRootParseNode($result);
+                    return $rootNode->getCollectionOfObjectValues($targetClass);
+                }
+                return $responseHandler->handleResponseAsync($result);
+            }
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function sendPrimitiveAsync(RequestInformation $requestInfo, string $primitiveType, ?ResponseHandler $responseHandler = null): Promise
+    {
+        $response = $this->getHttpResponseMessage($requestInfo);
+        return $response->then(
+            function (ResponseInterface $result) use ($primitiveType, $responseHandler) {
+                if (!$responseHandler) {
+                    $rootParseNode = $this->getRootParseNode($result);
+                    switch (strtolower($primitiveType)) {
+                        case 'int':
+                        case 'long':
+                            return $rootParseNode->getIntegerValue();
+                        case 'float':
+                            return $rootParseNode->getFloatValue();
+                        case 'bool':
+                            return $rootParseNode->getBooleanValue();
+                        case 'string':
+                            return $rootParseNode->getStringValue();
```

Comment:
Date, DateOnly, TimeOnly and Duration, see #1004

Reply:
Would the types above be considered primitive types? Or should they be handled by `sendAsync()` & `getObjectValue()` determines how to handle them?
(Same for collection equivalents of the same)

Reply:
considered primitives as sent as scalar strings over the wire, not complex objects

Reply:
> Date, DateOnly, TimeOnly and Duration, see #1004

Thinking the rest of this work may need to be done in a separate PR to reduce conflicts/duplication between Kenneth's PR and mine.